### PR TITLE
Add printHex() to OutBuffer

### PR DESCRIPTION
**Pull request by BorisCarvajal**
I removed this function from other pull, but still could be useful for some purpose.

### CODE REVIEW

File: src/dmd/root/outbuffer.d [hunk 1]
```diff
@@ -379,13 +380,29 @@ struct OutBuffer
      * Params:
      *  u = integral value to append
      */
-    extern (C++) void print(ulong u) pure nothrow
+    extern (C++) void print(uint64_t u) pure nothrow
```
Comment (Geod24) [inline src/dmd/root/outbuffer.d:383]: Why the change ?

**Comment by Geod24** [comment]
Couldn't one just use `printf` here ?

File: src/dmd/root/outbuffer.d (hunk 1 above)
Comment (BorisCarvajal) [reply src/dmd/root/outbuffer.d:383]: I've added a C++ usage of it, so I'm following the @ibuclaw [advise](https://github.com/dlang/dmd/pull/11555#discussion_r469192870).

**Comment by BorisCarvajal** [comment]
@Geod24 Well, `print` could use `printf` too, no?. But yeah maybe this function is useless after all.

File: src/dmd/root/outbuffer.d (hunk 1 above)
Comment (ibuclaw) [reply src/dmd/root/outbuffer.d:383]: Don't think that'd work on OSX.  You need a new `d_ulong` type similar to `d_size_t` that checks the size of `long`.

Comment (ibuclaw) [reply src/dmd/root/outbuffer.d:383]: Though I don't recall for how long the cpp_long fix has been in druntime for, but I think not long enough to fall within the period when dmd was broken for a few releases.

Comment (BorisCarvajal) [reply src/dmd/root/outbuffer.d:383]: Oh man, why something that simple is not yet handled easily.

**Comment by Geod24** [comment]
I would say it's not needed. Find if we close ?

**Comment by RazvanN7** [comment]
I'm closing this, feel free to reopen if new perspectives arise.
